controllers/capability/capability_configmap_controller.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024 Broadcom. All Rights Reserved.
+// Broadcom Confidential. The term "Broadcom" refers to Broadcom Inc.
+// and/or its subsidiaries.
+
+package capability
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+
+	pkgcfg "github.com/vmware-tanzu/vm-operator/pkg/config"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+	pkgmgr "github.com/vmware-tanzu/vm-operator/pkg/manager"
+	"github.com/vmware-tanzu/vm-operator/pkg/providers"
+	"github.com/vmware-tanzu/vm-operator/pkg/record"
+	kubeutil "github.com/vmware-tanzu/vm-operator/pkg/util/kube"
+)
+
+// AddToManager adds this package's controller to the provided manager.
+func AddToManager(ctx *pkgctx.ControllerManagerContext, mgr manager.Manager) error {
+	var (
+		controlledType      = &corev1.ConfigMap{}
+		controllerName      = "capability-configmap"
+		controllerNameShort = fmt.Sprintf("%s-controller", controllerName)
+		controllerNameLong  = fmt.Sprintf("%s/%s/%s", ctx.Namespace, ctx.Name, controllerNameShort)
+	)
+
+	r := NewReconciler(
+		ctx,
+		mgr.GetClient(),
+		ctrl.Log.WithName("controllers").WithName(controllerName),
+		record.New(mgr.GetEventRecorderFor(controllerNameLong)),
+		ctx.Namespace,
+		ctx.VMProvider,
+	)
+
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	cache, err := pkgmgr.NewNamespacedCacheForObject(
+		mgr,
+		&ctx.SyncPeriod,
+		controlledType,
+		WcpClusterCapabilitiesNamespace)
+	if err != nil {
+		return err
+	}
+
+	return c.Watch(source.Kind(
+		cache,
+		controlledType,
+		&handler.TypedEnqueueRequestForObject[*corev1.ConfigMap]{},
+		predicate.TypedFuncs[*corev1.ConfigMap]{
+			CreateFunc: func(e event.TypedCreateEvent[*corev1.ConfigMap]) bool {
+				return e.Object.GetName() == WcpClusterCapabilitiesConfigMapName
+			},
+			UpdateFunc: func(e event.TypedUpdateEvent[*corev1.ConfigMap]) bool {
+				return e.ObjectOld.GetName() == WcpClusterCapabilitiesConfigMapName
+			},
+			DeleteFunc: func(e event.TypedDeleteEvent[*corev1.ConfigMap]) bool {
+				return false
+			},
+			GenericFunc: func(e event.TypedGenericEvent[*corev1.ConfigMap]) bool {
+				return false
+			},
+		},
+		kubeutil.TypedResourceVersionChangedPredicate[*corev1.ConfigMap]{},
+	))
+}
+
+func NewReconciler(
+	ctx context.Context,
+	client client.Client,
+	logger logr.Logger,
+	recorder record.Recorder,
+	vmOpNamespace string,
```

Comment:
I do not see anywhere that `vmOpNamespace` is actually used. Did you intend to use it to limit the scope of the watch? If so, that was not done as best I can tell. See the `infra_configmap_controller.go` for an example. I think you want to initialize the cache for this controller with something like:

```go
	cache, err := pkgmgr.NewNamespacedCacheForObject(
		mgr,
		&ctx.SyncPeriod,
		controlledType,
		vmOpNamespace )
	if err != nil {
		return err
	}
```

Ultimately you do not need this parameter as part of `NewReconciler`, instead you need its value in `AddToManager`. Instead of passing it, you could just use a constant since the namespace should not be changing.